### DISCUSSION TITLE
Installing Next.js with Yarn Zero-Installs

### DISCUSSION THREAD

**Discussion by moiri-gamboni** [vercel/next.js#50648] [Help]
### Summary

When installing Next.js with `yarn create next-app` with default settings (next.js v13.4.4, yarn v3.5.1), `swc` and `node-gyp` are installed as unplugged packages. According to the instructions in [https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored):
> .yarn/unplugged should likely always be ignored since they typically hold machine-specific build artifacts. Ignoring it might however prevent [Zero-Installs](https://yarnpkg.com/features/zero-installs) from working (to prevent this, set [enableScripts](https://yarnpkg.com/configuration/yarnrc#enableScripts) to false).

If I ignore it, then zero-installs doesn't work as swc seems to be required (and it doesn't seem like it would be good to revert to Babel to fix this issue). 
```
$ yarn dev
- ready started server on 0.0.0.0:3000, url: http://localhost:3000
- warn Attempted to load @next/swc-linux-x64-gnu, but it was not installed
- warn Attempted to load @next/swc-linux-x64-gnux32, but it was not installed
- warn Attempted to load @next/swc-linux-x64-musl, but it was not installed
- error Failed to load SWC binary for linux/x64, see more info here: https://nextjs.org/docs/messages/failed-loading-swc
```
If I don't ignore it, I cannot commit it to GitHub as the file is too large:
```remote: error: See https://gh.io/lfs for more information.
remote: error: File my-app/.yarn/unplugged/@next-swc-linux-x64-gnu-npm-13.4.4-1f5fb12ec7/node_modules/@next/swc-linux-x64-gnu/next-swc.linux-x64-gnu.node is 122.42 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
```
I also tried setting `enableScripts: false` in `yarnrc.yml` but running `yarn install` again re-creates unplugged packages. 

Is there any way to fix this?

I found #27785, but it's quite old and has no replies. And also #48956, but it just suggests ignoring the unplugged directory.

**Comment by dannyrb** [comment]
Experiencing the same issue with the latest version of `.nextjs`, even when files are in the `.yarn/cache` compressed. The SWC dependencies are easily 3x to 4x larger than the next cached dependency, and by most others by significantly more.

**Comment by AlexandrNesterovSe7en** [comment]
я тоже получаю эту ошибку и не могу ничего сделать с этим
